calendar.js

```diff
@@ -9,16 +10,51 @@ function myTable() {
         table.appendChild(th);
     }
     //function to display hourly blocks of each day for one week
+    var time = getLengthsOfTimeArray();
+    
+    updateTable(table,time);
+    
+    document.body.appendChild(table);
+}
+myTable();
+
+function getLengthsOfTimeArray()
+{
+    var time = [];
+    $.ajax({
+        url: 'calendar.json',
+        dataType: 'json',
+        type: 'get',
+        cache: false,
+        success: function (data) {
+            $.each(data, function (i, item) {
+                time[i] = item.time.length;
+            })
+        }
+    })
+
+    return time;
```

Comment:
This will not work all the time because  `$.ajax ` is asynchronous. This means, before `time[i]` is updated in the `$.each` block, the function `updateTable` may be called with empty value which is a bug. The correct way to handle this logic, is to call `updateTable` *inside* the `success` callback of the `$.ajax` call. Please modify the code to this:
```
 success: function (data) {
   updateTable(data);
}
```
and update the logic in `updateTable` accordingly.